### PR TITLE
CV folds in HAL

### PR DESCRIPTION
**Pull request by rachaelvp**
Allow user to have control over cv.glmnet folds, instead of always passing `foldid` argument.

### CODE REVIEW

File: R/Lrnr_hal9001.R [hunk 1]
```diff
@@ -73,7 +73,28 @@ Lrnr_hal9001 <- R6Class(
       if (!any(grepl("fit_control", names(args)))) {
         args$fit_control <- list()
       }
-      args$fit_control$foldid <- origami::folds2foldvec(task$folds)
+
+      if (!any(grepl("fold", names(args$fit_control)))) {
+        args$fit_control$foldid <- origami::folds2foldvec(task$folds)
+      } else {
```
Comment (nhejazi) [inline R/Lrnr_hal9001.R:79]: @rachaelvp Could we clarify this part of the logic with some comments? I'm having trouble following exactly what happens here with stratifying on `Y` given the several conditional blocks

**Comment by nhejazi** [comment]
@rachaelvp, sorry I'm late to this, and I see that the PR has been merged already, but would you mind addressing the above comment when time allows? it'll be helpful to have a sense of what the intended logic was if/when we need to return to this later. also, please add some description of this changes to the `NEWS.md` file

**Comment by rachaelvp** [comment]
I addressed these comments in an open PR here: https://github.com/tlverse/sl3/pull/385. I ended up creating a task function `get_folds` so there's some uniformity in the construction of CV folds for learners that use CV to select tuning parameters (`Lrnr_caret`, `Lrnr_sl`, `Lrnr_hal9001`, and `Lrnr_glmnet`). Also if you can think of other learners I missed that also use CV to select tuning parameters please let me know.

**Comment by nhejazi** [comment]
thanks for addressing these points; i think enforcing uniformity in how CV gets handled between different learners is definitely very important. i’ll go thru the learner list and see if i can think of any others, but it also occurs to me that we may want to add “cv” as part of the learner tags as we’re doing this

**Pull request merged**